bridgit/InterractiveApp.cs

```diff
@@ -1,62 +1,331 @@
+        if (fileName == "") fileName = "Open";
+
+        Console.Write($"Filename (blank for default \"{fileName}\"): ");
+        var input = Console.ReadLine();
+        if (!string.IsNullOrEmpty(input)) fileName = input;
+
+        if (GameFile.FileExists(TestDirectory, fileName))
+        {
+            if (!Confirm("File already exists.  Overwrite (Y/N)? "))
+            {
+                return;
+            }
+        }
+        var gameFile = GameFile.NewGame(TestDirectory, fileName);
+
+        if (desiredCalls.Count == 0)
+        {
+            for (int i = 0; i < numTests; i++)
+            {
+                gameFile.Add(CreateTest.NewTest(i+1, singleOnly, initialAuction, null, completeAuction));
+            }
+        }
+        else
+        {
+            int boardNumber = 1;
+            foreach (var call in desiredCalls)
+            {
+                for (int i = 0; i < numTests; i++)
+                {                
+                    gameFile.Add(CreateTest.NewTest(boardNumber, singleOnly, initialAuction, call, completeAuction));
+                    boardNumber++;
+                }
+            }
+        }
+
+        this._failedTests = new Game[0];
+        this._gameFile = gameFile;
+        this._selectedGame = 0;
+        EditGameFile();
+    }
+
+    private void LoadTests()
+    {
+        Console.Clear();
+        var gameFiles = GameFile.EnumDirectory(TestDirectory);
+        for (int i = 0; i < gameFiles.Length; i++)
+        {
+            Console.WriteLine($"{i + 1, 3}: {gameFiles[i].FileName}");
+        }
+        Console.WriteLine();
+        Console.Write("Number of file to load or 0 to exit: ");
+        while (true)
+        {
+            var input = Console.ReadLine();
+            if (input == null) input = "";
+            int selected;
+            if (int.TryParse(input, out selected) && selected >= 0 && selected <= gameFiles.Length + 1)
             {
-                ProcessPbnText(input);
+                if (selected == 0) return;
+                this._gameFile = gameFiles[selected - 1];
+                this._gameFile.Load();
+                this._selectedGame = 0;
+                this._failedTests = TestEditor.FailingTests(_gameFile);
+                EditGameFile();
+                return;
             }
-            else
+        }
+    }
+
+    private void EditGameFile()
+    {
+        while (true)
+        {
+            Console.Clear();
+            if (_failedTests.Length > 0)
             {
-                Console.WriteLine($"Unknonwn command {input}");
+                Console.ForegroundColor = ConsoleColor.Red;
+                Console.WriteLine($"{_failedTests.Length} auction are failing.");
+                Console.ResetColor();
             }
+            Console.WriteLine($"Editing {_gameFile.FileName}, Board {_gameFile[_selectedGame].Board}, {_selectedGame + 1} of {_gameFile.Count}");
```

Comment:
The build is generating some possible `null` reference warnings you may want to resolve here.